### ISSUE TITLE
isettools/sceneEye changes Part I Issue: sceneEye TODO's

### ISSUE THREAD

**Issue by jenmaxwell** [isetbio/isetbio#291]
Assign some person(s) to handle the To-Do list for this function. Address the list of notes from the function.
- [ ] TODO - Implement the check that BW describes below. Is there a way to check inputs? For example, eyePos is not a dependent variable, but is instead read in from the PBRT file. However, say the user wants to change the value in their script so they write: myScene = sceneEye('pbrtFile', xxx); myScene.eyePos = [x y z]; Is there a way to ensure they put in a 3x1 vector for eyePos, other than just rigourous error checking in the code?
- [ ] [Note: BW - (Reference first TODO) Yes, I think so, using the myScene.set('eyePos', val) approach, or perhaps myScene.set.eyePos = val. In these cases the set operation can pass through an input parser that validates the input value (I think).]
- [ ] [Note: BW - I wonder if recipe should be a slot in here or whether we should use myScene.render(recipe, varargin); The current way does make sense, since this is actually a scene.]
- [ ] [Note: BW - maxDepth & nWaveBands are often empty, so let's perform the checks below. However, I should find a more permanant solution to cases like these. (See the note above). Maybe in piGetRenderRecipe we should put in the default values if any of these rendering options are missing (e.g. if Renderer is missing, put in Renderer 'sampler'.)]
- [ ] [Note: XXX - Actually...should we include this? This will change depending on the accommodation...- Public Get, Private Set section]
- [ ] TODO - Look (Constant property wave) up and fill it in.
- [ ] [Note: XXX - (from constructor) What happens if the recipe doesn't include any of the following, or any of the subfields we call?]
- [ ] TODO - Determine a better way to infer the accommodation. Currently we assume the naming conventions is "%s_%f.dat" This is not foolproof, so maybe we can think of a more robust way to do this in the future?
- [ ] [Note: TL - What does this hiddenHandle mean? I seem to need it to avoid errors.]
- [ ] TODO - Fix example!
- [x] [Note: JNM - There are a couple places that say rendered images are always square. Is this intentionally done, or something that needs to be fixed?]

**Comment by tlian7** [comment]
Hi @jenmaxwell,

Thanks for doing all of this! Since SceneEye is primarily mine, I know that some of these are minor fixes I can make when I get some time. If I do complete any of these "TODO's", should I make a comment here so we can cross it off? 

For example:

> [Note: JNM - There are a couple places that say rendered images are always square. Is this intentionally done, or something that needs to be fixed?]

Yes, this is done intentionally since this is how the rendering samples from the back of the retina. In other words, at the moment you can _only_ render square retinal images using sceneEye.

**Comment by render-toolbox** [comment]
Trisha et al.

I'd say that in a case like this, the way to proceed is

1) Add a sentence to the description section of the header comments that indicates that the rendered images can only be square because of the way the renderer is implemented.

2) Delete the note from the source.  If you want, add a line to the history section that you handled and deleted a note.

3) Find the issue relevant to the note, comment that it's handled in the source, and close.

4) Say something sensible in the commit comment about what you did.

More generally, if you address an issue raised by a note in the source, delete the note, search for an issue, and if there is an issue close it.

To put it another way, our goal is to get rid of all the "[Note: XXX -" comments by handling them, and not to keep a record in the source of how they are handled.

Best,

David

**Comment by jenmaxwell** [comment]
Trisha,

I think what we’ve discussed before is that we just remove the note/todo once it’s complete.

Jenn

**Comment by jenmaxwell** [comment]
Update 11/27 to convert to task format.

@tlian7 you can now check off items as they are complete (and reference issue in check-in notes!) 😃
